### ISSUE TITLE
Enforce memory limits on extensions

### ISSUE THREAD

**Issue by tvvignesh** [microsoft/vscode#175249]
I just figured out that there is a memory leak with `lit-plugin` extension and I reported it here: https://github.com/runem/lit-analyzer/issues/298

This ended up taking up my complete RAM in my remote VM (I develop using remote SSH) leading to the VM rendering itself unresponsive and I had to reboot the VM again.

Now, while the reason for the memory leak is an issue with the extension, my concern is that VSCode must not allow extensions for consuming such huge amounts of memory. I already discussed this here but bringing this back up again since its critical to be addressed: https://github.com/microsoft/vscode/issues/127728#issuecomment-872663985

In summary, this is a feature request for VSCode to allow for throttling of resources used by extensions (with limits automatically set / specified by the user) - this will avoid such issues. Thanks.

**Comment by alexdima** [comment]
This can only be possible if we implement a sandbox for running extensions. Let's track in a very similar issue -- https://github.com/microsoft/vscode/issues/52116
